### DISCUSSION TITLE
X.509 certificate Tutorial

### DISCUSSION THREAD

**Discussion by chandra-rayaprol** [Azure/azure-iot-sdk-csharp#3172] [Q&A] [answered]
I'm just testing out this tutorial "https://learn.microsoft.com/en-us/azure/iot-hub/tutorial-x509-test-certificate" and would like to update to the latest version ''Microsoft.Azure.Devices.Client 2.0.0-preview004" as a dependency.  I could not find the class equivalent to "DeviceAuthenticationWithX509Certificate". Could you kindly recommend the code modifications needed to compile and build with the V2 library?

**Comment by drwill-ms** [comment, marked as answer]
Hi @Raysapps -

Thank you for trying out the v2 preview!

The renamed type you are looking for is `ClientAuthenticationWithX509Certificate`. We squashed the one for a device and the one for a module into a single class.

We have a similar sample written for v2 [here](https://github.com/Azure/azure-iot-sdk-csharp/tree/previews/v2/iothub/device/samples/how%20to%20guides/X509DeviceCertWithChainSample).

I see this mapping is missing from our [migration guide](https://github.com/Azure/azure-iot-sdk-csharp/blob/previews/v2/SDK%20v2%20migration%20guide.md), so I'll ensure it gets added.

Regards,
David

**Comment by chandra-rayaprol** [comment]
Hi @drwill-ms,
Thank you for sharing the relevant V2 class and the sample. Everything is working fine.
Regards,
Chandra Mohan
